### PR TITLE
chore: replace debezium in ci and test on multiple pg versions

### PR DESCRIPTION
**Pull request by iamKunalGupta**

### CODE REVIEW

**Comment by iamKunalGupta** [comment]
once approved, I will update the merge rules to fix the status check requirement

**Review by serprex — APPROVED** [review]

File: .github/workflows/flow.yml [hunk 1]
```diff
@@ -11,11 +11,12 @@ jobs:
     strategy:
       matrix:
         runner: [ubicloud-standard-16-ubuntu-2204-arm]
+        postgres-version: [15, 16, 17]
     runs-on: ${{ matrix.runner }}
     timeout-minutes: 30
     services:
       catalog:
-        image: imresamu/postgis:15-3.4-alpine
+        image: imresamu/postgis:${{ matrix.postgres-version }}-3.4-alpine
```
Comment (heavycrystal) [inline .github/workflows/flow.yml:19]: I see PostGIS 3.5

Comment (iamKunalGupta) [reply .github/workflows/flow.yml:19]: I did not update the image version as I was not sure

Comment (iamKunalGupta) [reply .github/workflows/flow.yml:19]: Was waiting for tests to pass firt, will update this as well

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,18 +5,19 @@ on:
...
     branches: [main, release/*]
-    paths: [nexus/**, protos/**]
+    paths: [nexus/**, protos/**, .github/workflows/ci.yml]
 
 jobs:
   build:
     strategy:
       matrix:
         runner: [ubicloud-standard-2-ubuntu-2204-arm]
+        postgres-version: [14, 15, 16, 17]
```
Comment (heavycrystal) [inline .github/workflows/ci.yml:15]: can add PG 13 here as well; 12 will be EOL soon

Comment (iamKunalGupta) [reply .github/workflows/ci.yml:15]: we have customers from pg 13?

File: .github/workflows/flow.yml [hunk 3]
```diff
@@ -11,11 +11,12 @@ jobs:
     strategy:
       matrix:
         runner: [ubicloud-standard-16-ubuntu-2204-arm]
+        postgres-version: [15, 16, 17]
```
Comment (heavycrystal) [inline .github/workflows/flow.yml:14]: we need only run a subset of tests ideally; PG-PG and some tests to ensure initial load/resync works; fine for now
13 and 14 should be here but they're pre-MERGE so will need work

File: .github/workflows/ci.yml (hunk 2 above)
Comment (heavycrystal) [reply .github/workflows/ci.yml:15]: we allow PG13 peers to be created

File: .github/workflows/flow.yml (hunk 3 above)
Comment (iamKunalGupta) [reply .github/workflows/flow.yml:14]: can you create an issue for now @heavycrystal ?

**Pull request merged**